### ISSUE TITLE
pyright uses system locale regardless of envvars

### ISSUE THREAD

**Issue by denballakh** [microsoft/pyright#7439]
**Describe the bug**
Documentation says:
> By default, pyright uses the default locale of the operating system. You can override the desired locale through the use of one of the following environment variables, listed in priority order.
```
LC_ALL="de"
LC_MESSAGES="en-us"
LANG="zh-cn"
LANGUAGE="fr"
```

Actually, pyright doesnt respect these environment variables. 
No locale set, random one is used: 
```bat
D:\>echo %LC_ALL% %LC_MESSAGES% %LANG% %LANGUAGE%
%LC_ALL% %LC_MESSAGES% %LANG% %LANGUAGE%

D:\>pyright _.py
d:\_.py
  d:\_.py:1:10 - error: Выражение типа "Literal['']" нельзя присвоить объявленному типу "int"
    "Literal['']" несовместим с "int" (reportAssignmentType)
1 error, 0 warnings, 0 informations
```
Lets set something manually: 
```bat
D:\>set LC_ALL=en

D:\>set LC_MESSAGES=en-us

D:\>set LANG=en-us

D:\>set LANGUAGE=en

D:\>echo %LC_ALL% %LC_MESSAGES% %LANG% %LANGUAGE%
en en-us en-us en

D:\>pyright _.py
d:\_.py
  d:\_.py:1:10 - error: Выражение типа "Literal['']" нельзя присвоить объявленному типу "int"
    "Literal['']" несовместим с "int" (reportAssignmentType)
1 error, 0 warnings, 0 informations
```
No difference. 

**Expected behaviour**
The english locale is used by default, unless something else is specified explicitly.
Non-english error messages are poorly translated. They make it difficult to search for the error, because vast majority of internet speaks english, and they make it more difficult to share the errors with other people who dont know the language that is set in my OS. 

**Some info**
- Windows 10
- CPython3.12
- pyright `1.1.353`

**Comment by denballakh** [comment]
update: i set my system locale to english, rebooted. Error messages are still in russian. Reinstalling pyright didnt help

**Comment by erictraut** [comment]
From your report, there are four separate issues:
1. You would like pyright to default to english rather than the system locale. This is by design, and we're unlikely to change it because other pyright users prefer the current behavior.
2. Some of the strings are translated poorly. @bschnurr, I've heard this from other people as well. Is there a way the pylance team could supply better translations? If the current translations are really bad for a particular language, then it's probably best to just remove them and fall back to english for that locale.
3. You tried to use the documented environment variables to override the system locale, and it didn't appear to work. I've tried to repro this, and it appears to be working fine for me. This might be a platform-specific issue though. I can confirm that it's working on MacOS and Linux, but I don't have ready access to a Windows machine. @bschnurr, could you verify that this is working on Windows?
4. You have changed your system locale, and pyright didn't pick up the change the next time you ran it. Once again, I'm unable to repro this on MacOS or Linux. If you're seeing Russian strings, then pyright is picking up that locale from somewhere — either your system locale or from environment variables that are set on your system.

**Comment by denballakh** [comment]
1. why is there no CLI option to force desired language for messages?
2. why are you adding bad and confusing (probably auto-generated) messages and, most importantly, forcing users to use this trash?
4. i also removed the cached folder that contained javascript stuff, on the next launch pyright installed it again, but it didn't help. I do not have any of the mentioned envvars set

i deobfuscated your js code, and figured out that locale is returned by this code: 
```ts
    try {
        if (navigator?.language) {
            return navigator.language.toLowerCase();
        }
    } catch {
        // Fall through
    }
```
what is this? i dont see much code that touches `navigator`
why does it have higher priority than envvars?
why is there no way to force pyright use specific language?

**Comment by denballakh** [comment]
I downgraded nodejs to v20, and it fixed the issue.
You are using `navigator?.language` for some unknown reason, and it takes priority over all other ways to specify locale.

**Comment by erictraut** [comment]
The pyright code base is used for a browser-based tool (vscode.dev). The `navigator?.language` reference was intended to be applicable only in browser environments. It sounds from your analysis that versions of node > v20 have added support for `navigator`. @bschnurr, could you please update the logic you added for detecting the default locale? The current technique will no longer work. Perhaps you could abstract out the `navigator` reference so it applies only when you know that you're running in a browser environment.

**Comment by erictraut** [comment]
@bschnurr, since I didn't hear from you, I went ahead and removed the code that looks at `navigator?.language`. This is a pretty significant regression for pyright users who update to node 21, so I think it's important to get this change into this week's update. If pylance needs to continue to work in a browser and you want to leverage `navigator?.language` to select the default language in that environment, you'll need to find a better way to abstract that logic so it runs only in the browser.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.354, which I just published.
